context_lab/src/components/Weather.js

```diff
@@ -0,0 +1,83 @@
+import React, { useState, useEffect } from "react";
+import "../css/Weather.css";
+import apiKey from "../secret";
+import axios from "axios";
+import WeatherDisplay from "./WeatherDisplay";
+
+export const farenheit = temp => {
+  return (temp - 273.15) * 1.8 + 32;
+};
+
+const Weather = () => {
+  const [location, setLocation] = useState([]);
+  const [temperature, setTemperature] = useState([]);
+  const [weather, setWeather] = useState([]);
+  const [feelsLike, setFeelsLike] = useState("");
+  const [uvi, setUvi] = useState("");
+  const [dew, setDew] = useState("");
+  const [pressure, setPressure] = useState("");
+  const [humidity, setHumidity] = useState("");
+
+  const getAddress = async () => {
+    await navigator.geolocation;
+    navigator.geolocation.getCurrentPosition(position => {
+      let lat = position.coords.latitude;
+      let lon = position.coords.longitude;
+      setLocation([lat, lon]);
+    });
+  };
+
+  useEffect(() => {
+    getAddress();
+  }, []);
+
+  let lat = location[0];
+  let lon = location[1];
+
+  const getWeather = async () => {
+    try {
+      let realTemp = 0;
+      let res = await axios.get(`https://api.openweathermap.org/data/2.5/onecall?lat=${lat}&lon=${lon}&appid=${apiKey}
+      `);
+      const stat = res.data.current;
+      let temp = stat.temp;
+      realTemp = (Math.round(farenheit(temp)));
+      setTemperature(realTemp);
```

Comment:
to create just one state object here by first defining a variable ex: `let newweatherObj = {}`

then just add a key for each of these states like `newweatherObj.description = res.data.current.weather[0].description`
then just setting that object `setWeatherObj(newweatherObj)`